File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const gym = new Array(n).fill(1);
    for (let i of lost){
        gym[i-1]--;
    }
    for (let i of reserve){
        gym[i-1]++;
    }
    for (let i=0; i<n; i++){
        if (gym[i]==0){
            if (i-1>=0 && gym[i-1]>1){
                gym[i-1]--;
                gym[i]++;
            } else if (i+1<n && gym[i+1]>1){
                gym[i+1]--;
                gym[i]++;
            }
        }
    }
    let cnt= 0;
    for (let i of gym){
        if (i>=1) cnt++;
    }
    return cnt;
}

File: 프로그래머스/1/42862. 체육복/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        
        int[] result = new int[n];
        for (int i=0; i<n; i++){
            result[i] = 1;
        }
        for (int l : lost){
            result[l-1] -=1;
        }
        for (int r : reserve) {
            result[r-1] += 1;
        }
        
        for (int i=0; i<n; i++){
            if (result[i]==0){
                if (i-1>=0 && result[i-1]>1){
                    result[i-1]--;
                    result[i]++;
                } else if (i+1<n && result[i+1]>1){
                    result[i+1]--;
                    result[i]++;
                }
            }
        }
        
        for (int i=0; i<n; i++){
            if (result[i]!=0) answer++;
        }
        
        return answer;
    }
}